pkg/minikube/machine/build_images.go

```diff
@@ -59,6 +61,14 @@ func BuildImage(path string, file string, tag string, push bool, env []string, o
 		remote = false
 	}
 
+	if tag != "" {
+		named, err := dockerref.ParseNormalizedNamed(tag)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't parse image reference %q", tag)
+		}
+		tag = named.Name()
```

Comment:
This line is the incorrect line, which makes test failes

for example, if tag="localhost/my-image:minikube", named.Name() returns "localhost/my-image"

named.String() returns the full reference.
```suggestion
		tag = named.String()
```

Reply:
Sounds about right, I probably aonly tested with ":latest" (which works either way)